src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -276,6 +276,29 @@ protected void CheckEffectiveAccessibility(TypeWithAnnotations returnType, Immut
             diagnostics.Add(Locations[0], useSiteInfo);
         }
 
+        protected void CheckFileTypeUsage(TypeWithAnnotations returnType, ImmutableArray<ParameterSymbol> parameters, BindingDiagnosticBag diagnostics)
+        {
+            if (ContainingType.IsFileTypeOrUsesFileTypes())
+            {
+                return;
+            }
+
+            if (returnType.Type.IsFileTypeOrUsesFileTypes())
+            {
+                diagnostics.Add(ErrorCode.ERR_FileTypeDisallowedInSignature, Locations[0], returnType.Type, ContainingType);
+                return;
```

Comment:
>return [](http://example.com/codeflow?start=16&length=6)

Consider continuing to check parameters rather than returning, similar to `CheckEffectiveAccessibility()`. (If there are two distinct types across return type and parameters that are `file` types, we should report errors for both, and it seems reasonable to me to report multiple errors even if the types are the same.)